### ISSUE TITLE
Android app rarely enables scraping

### ISSUE THREAD

**Issue by galmok** [lephisto/tesla-apiscraper#66]
My experience with the Android app is that while I usually get it to disable scraping automatically, it rarely manages to enable automatically. Is there something I can do to make the Android app more reliable. Often I find I have collected no data of my trip and that is becoming an irritation for me.

I am using the app on my Samsung $9, and while it has lots of battery saving features, I believe I have disabled all of them for this app. I have another app that reliably detects  its Bluetooth device coming online, and I didn't have to configure anything for that app (it is Saphe). So I know it is possible to do. :-) I just don't know how...

**Comment by galmok** [comment]
Looking closer, I notice that Saphe uses a service. That service is probably necessary to reliably detect changes in Bluetooth. The Tesla API Scraper app is unfortunately unreliable.

**Comment by lephisto** [comment]
Often the problem is caused by the fact that when en/disable wifi disconnect happens as wellt, and the Android Retrypolicy is not working so great. But I'm far from being an Android expert, feel free to contribute here: https://github.com/lephisto/tesla-apiscraper-app

What always works is setting it manually and getting confirmation of scraper is on or off.

**Comment by Vipercat** [comment]
Apologies for the dumb question, but do I need to open certain ports on the server where I'm running the Apiscraper for the app to connect to it? In the App I've now put in the URL based on http://<dns name> and of course the API key that I've also put in the config file on the server.

**Comment by galmok** [comment]
Yes. The configured port must be opened in your router and be forwarded to
the machine the had the apiscraper running.

lør. 31. aug. 2019 17.43 skrev Vipercat <notifications@github.com>:

**Comment by Vipercat** [comment]
And what port is that?

**Comment by galmok** [comment]
According to the config file:

a_api_port = 8023

**Comment by Vipercat** [comment]
Yep, it works. Thanks!
